src/components/versionista-info.jsx

```diff
@@ -19,55 +19,70 @@ import {dateFormatter} from '../scripts/formatters';
  */
 export default class VersionistaInfo extends React.Component {
   render () {
-    let message;
     if (!this.props.from
      || !this.props.to
      || this.props.from.source_type !== 'versionista'
      || this.props.to.source_type !== 'versionista') {
       return null;
     }
-    else if (this.props.from === this.props.to) {
-      message = <span>There is only one version. No diff to display.</span>;
+
+    return (
+      <div className="versionista-info">
+        {this.getMessage() || this.renderDiffLink()}
+      </div>
+    );
+  }
+
+  getMessage () {
+    let message = null;
+    if (this.props.from === this.props.to
+      || this.props.to.uuid === this.props.from.uuid) {
+      message = this.renderPageLink();
     }
     else {
       message = this.renderMissingVersionsMessage();
     }
+    return message;
```

Comment:
Since you aren’t doing anything with `message` after obtaining it here, you may as well just ditch the variable altogether and:

```js
getMessage () {
  if (this.props.to.uuid === this.props.from.uuid) {
    return this.renderPageLink();
  }
  else {
    return this.renderMissingVersionsMessage();
  }
}
```

Reply:
Ahh, that's true. In my mind, it's not immediately obvious that `getMessage` can return `null`, so stating it explicitly doesn't raise a code smell from this [line](https://github.com/lightandluck/web-monitoring-ui/blob/b8875f3ba4c4ec241be18dbae31a861a470b1fa7/src/components/versionista-info.jsx#L31) if I was to re-read the code. I know this component is temporary, but for more permanent code, is this worth considering or am I just overly cautious and underestimating future me? 😅

Reply:
I hope you are underestimating future you :)

I hear what you’re saying, but the fact that you have an `else` clause with no conditions means `message` will never not be the result of one of these two method calls. I think it’s better, in the end, to make *that* clearer here.

Another approach—less pretty, but maybe more clear—would be to hoist the logic of whether there are missing versions to render a message about up from `renderMissingVersionsMessage()` up into this method, e.g:

```js
getMessage () {
  if (this.props.to.uuid === this.props.from.uuid) {
    return this.renderPageLink();
  }
  
  const missingVersions = this.getDatesWithoutDiff();
  if (missingVersions) {
    return this.renderMissingVersionsMessage(missingVersions);
  }

  return null;
}
```

Reply:
A part of me did want to use the alternative you suggested, but I'm just gonna trust and go with the more succinct and clearer version. Thanks for your help!